framework/src/st_things/things_stack/utils/things_ping.c

```diff
@@ -69,43 +74,197 @@ typedef struct things_ping_s {
+ * @param client_response : response from peer.
+ * @return OCStackApplicationResult : application result.
+ */
 static OCStackApplicationResult discover_ping_resource_handler(void *ctx, OCDoHandle handle, OCClientResponse *client_response);
+
+//--------------------------------------------------------------------------------
+/**
+ * @brief Static type prototype for handle ping request.
+ * @param ctx : context value
+ * @param handle : Handle to an OCDoResource invocation.
+ * @param client_response : response from peer.
+ * @return OCStackApplicationResult : application result.
+ */
 static OCStackApplicationResult handler_ping_request(void *ctx, OCDoHandle handle, OCClientResponse *client_response);
 
+//--------------------------------------------------------------------------------
+/**
+ * @brief Static type prototype for creating things_ping_s.
+ * @param remote_addr : Remote address
+ * @param port : port number
+ * @return structure of things_ping_s
+ */
 static things_ping_s *create_things_ping_s(const char *remote_addr, const uint16_t port);
+
+//--------------------------------------------------------------------------------
+/**
+ * @brief State type prototype for terminate things_ping_s.
+ * @param ping : Structure of things_ping_s
+ */
 static void terminate_things_ping_s(things_ping_s *ping);
+
+//--------------------------------------------------------------------------------
+/**
+ * @brief Static type prototype for matching ip.
+ * @param ping : Structure of things_ping_s
+ * @param target_ip : IP of target device.
+ * @return success = 1, otherwise 0.
+ */
 static int is_ip_key_equal(things_ping_s *ping, const char *target_ip);
+
+//--------------------------------------------------------------------------------
+/**
+ * @brief Static type prototype for checking ping thread. Creating ping thread or
+ * destroy ping thread like that.
+ * @param ping : Structure of things_ping_s
+ */
 static void check_ping_thread(things_ping_s *ping);
+
+//--------------------------------------------------------------------------------
+/**
+ * @biref Static type prototype for set the state of Ping of things_ping_s.
+ * @param ping : Structure of things_ping_s
+ * @param state : State of Ping of things_ping_s
+ */
 static void set_mask(things_ping_s *ping, ping_state_e state);
+
+//--------------------------------------------------------------------------------
+/**
+ * @biref Static type prototype for get the state of Ping of things_ping_s.
+ * @param ping : Structure of things_ping_s
+ * @param state : State of Ping of things_ping_s
+ * @return success = true, failure = false.
+ */
 static bool get_mask(things_ping_s *ping, ping_state_e state);
+
+//--------------------------------------------------------------------------------
+/**
+ * @biref Static type prototype for set the state of Ping of things_ping_s.
+ * @param ping : Structure of things_ping_s
+ * @param state : State of Ping of things_ping_s
```

Comment:
state is not this param description. 
and you need to add preStateThen variable description.
please check it.

Reply:
modified.